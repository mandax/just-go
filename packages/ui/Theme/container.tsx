import { CSSProperties } from "react";
import theme from "./index";
import { rem } from "./units";
import { font } from "./font";

export type ContainerGenerator =
  (hPadding?: number, vPadding?: number) => React.CSSProperties

export const containerBase = (
  fontColor: string = theme.COLOR_PRIMARY,
  bgColor: string = theme.COLOR_BASE,
): ContainerGenerator => (
  hPadding: number = theme.DEFAULT_HORIZONTAL_PADDING,
  vPadding: number = theme.DEFAULT_VERTICAL_PADDING
): CSSProperties => ({
  ...font(),
  color: fontColor,
  backgroundColor: bgColor,
  padding: `${rem(vPadding)} ${rem(hPadding)}`
});

export const container: ContainerGenerator = containerBase();

export const containerTransparent: ContainerGenerator = 
  containerBase(theme.COLOR_PRIMARY, 'transparent');

export const containerAccent: ContainerGenerator = 
  containerBase(theme.COLOR_BASE, theme.COLOR_PRIMARY);

export const verticalCenter: React.CSSProperties = {
  position: 'absolute',
  top: '50%',
  transform: 'translateY(-50%)'
};

export const horizontalCenter: React.CSSProperties = {
  margin: `0 auto`
};

export const roundedBorder = (
  defaultOrtopLeftBorder: number = theme.ROUNDED_BORDER,
  topRightBorder?: number,
  bottomLeftBorder?: number,
  bottomRightBorder?: number
):React.CSSProperties => {
  let css:React.CSSProperties = {
    borderRadius: defaultOrtopLeftBorder
  };
  if (topRightBorder || bottomLeftBorder || bottomRightBorder) {
    css = {
      borderTopLeftRadius: defaultOrtopLeftBorder,
      borderTopRightRadius: topRightBorder,
      borderBottomLeftRadius: bottomLeftBorder,
      borderBottomRightRadius: bottomRightBorder
    }
  }
  return css;
};

