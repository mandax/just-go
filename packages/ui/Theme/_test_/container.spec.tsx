import * as ctnr from "../container";

describe('helper container', () => {
    
  it('should generate a default container', () => {
    expect(ctnr.containerBase()()).toMatchSnapshot()
  })
   
  it('should generate a new color container', () => {
    expect(ctnr.containerBase('red', 'blue')()).toMatchSnapshot()
  })
   
  it('should generate a new top/bottom padding container', () => {
    expect(ctnr.containerBase()(undefined, 2)).toMatchSnapshot()
  })
  
  it('should generate a new side padding container', () => {
    expect(ctnr.containerBase()(2)).toMatchSnapshot()
  })

  it('should return a default css container', () => {
    expect(ctnr.container()).toMatchSnapshot()
  })
  
  it('should return a transparent background css container', () => {
    expect(ctnr.containerTransparent()).toMatchSnapshot()
  })
  
  it('should return a accent css container', () => {
    expect(ctnr.containerAccent()).toMatchSnapshot()
  })

  it('should return a vertical centralized container css', () => {
    expect(ctnr.verticalCenter).toMatchSnapshot()
  })
  
  it('should return a horizontal centralized container css', () => {
    expect(ctnr.horizontalCenter).toMatchSnapshot()
  })
  
  it('should return a rounded border css container', () => {
    expect(ctnr.roundedBorder()).toMatchSnapshot()
  })

  it('should return only top left rounded border css container', () => {
    expect(ctnr.roundedBorder(10)).toMatchSnapshot()
  })
  
  it('should return only top right rounded border css container', () => {
    expect(ctnr.roundedBorder(undefined, 10)).toMatchSnapshot()
  })
  
  it('should return only bottom left rounded border css container', () => {
    expect(ctnr.roundedBorder(undefined, undefined, 10)).toMatchSnapshot()
  })

  it('should return only bottom right rounded border css container', () => {
    expect(ctnr.roundedBorder(undefined, undefined, undefined, 10)).toMatchSnapshot()
  })
})