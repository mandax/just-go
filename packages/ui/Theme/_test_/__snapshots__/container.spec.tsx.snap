// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helper container should generate a default container 1`] = `
Object {
  "backgroundColor": "#FFFFFF",
  "color": "#263238",
  "fontFamily": "Manjari",
  "fontSize": "1rem",
  "fontWeight": 400,
  "padding": "2rem 3rem",
}
`;

exports[`helper container should generate a new color container 1`] = `
Object {
  "backgroundColor": "blue",
  "color": "red",
  "fontFamily": "Manjari",
  "fontSize": "1rem",
  "fontWeight": 400,
  "padding": "2rem 3rem",
}
`;

exports[`helper container should generate a new side padding container 1`] = `
Object {
  "backgroundColor": "#FFFFFF",
  "color": "#263238",
  "fontFamily": "Manjari",
  "fontSize": "1rem",
  "fontWeight": 400,
  "padding": "2rem 2rem",
}
`;

exports[`helper container should generate a new top/bottom padding container 1`] = `
Object {
  "backgroundColor": "#FFFFFF",
  "color": "#263238",
  "fontFamily": "Manjari",
  "fontSize": "1rem",
  "fontWeight": 400,
  "padding": "2rem 3rem",
}
`;

exports[`helper container should return a accent css container 1`] = `
Object {
  "backgroundColor": "#263238",
  "color": "#FFFFFF",
  "fontFamily": "Manjari",
  "fontSize": "1rem",
  "fontWeight": 400,
  "padding": "2rem 3rem",
}
`;

exports[`helper container should return a default css container 1`] = `
Object {
  "backgroundColor": "#FFFFFF",
  "color": "#263238",
  "fontFamily": "Manjari",
  "fontSize": "1rem",
  "fontWeight": 400,
  "padding": "2rem 3rem",
}
`;

exports[`helper container should return a horizontal centralized container css 1`] = `
Object {
  "margin": "0 auto",
}
`;

exports[`helper container should return a rounded border css container 1`] = `
Object {
  "borderRadius": 4,
}
`;

exports[`helper container should return a transparent background css container 1`] = `
Object {
  "backgroundColor": "transparent",
  "color": "#263238",
  "fontFamily": "Manjari",
  "fontSize": "1rem",
  "fontWeight": 400,
  "padding": "2rem 3rem",
}
`;

exports[`helper container should return a vertical centralized container css 1`] = `
Object {
  "position": "absolute",
  "top": "50%",
  "transform": "translateY(-50%)",
}
`;

exports[`helper container should return only bottom left rounded border css container 1`] = `
Object {
  "borderBottomLeftRadius": 10,
  "borderBottomRightRadius": undefined,
  "borderTopLeftRadius": 4,
  "borderTopRightRadius": undefined,
}
`;

exports[`helper container should return only bottom right rounded border css container 1`] = `
Object {
  "borderBottomLeftRadius": undefined,
  "borderBottomRightRadius": 10,
  "borderTopLeftRadius": 4,
  "borderTopRightRadius": undefined,
}
`;

exports[`helper container should return only top left rounded border css container 1`] = `
Object {
  "borderRadius": 10,
}
`;

exports[`helper container should return only top right rounded border css container 1`] = `
Object {
  "borderBottomLeftRadius": undefined,
  "borderBottomRightRadius": undefined,
  "borderTopLeftRadius": 4,
  "borderTopRightRadius": 10,
}
`;
